uni/lib/view/lazy_consumer.dart

```diff
@@ -19,28 +19,34 @@ class LazyConsumer<T extends StateProviderNotifier> extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    try {
-      final sessionProvider = Provider.of<SessionProvider>(context);
-      final profileProvider = Provider.of<ProfileProvider>(context);
-
-      WidgetsBinding.instance.addPostFrameCallback((_) async {
-        final session = sessionProvider.session;
-        final profile = profileProvider.profile;
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      try {
+        // Load data stored in the database immediately
         final provider = Provider.of<T>(context, listen: false);
+        await provider.ensureInitializedFromStorage();
 
+        // If the provider fetchers depend on the session, make sure that
+        // SessionProvider and ProfileProvider are initialized
         if (provider.dependsOnSession) {
-          sessionProvider.ensureInitialized(session, profile).then((_) =>
-              profileProvider
-                  .ensureInitialized(session, profile)
-                  .then((_) => provider.ensureInitialized(session, profile)));
-        } else {
-          provider.ensureInitialized(session, profile);
+          if (context.mounted) {
+            await Provider.of<SessionProvider>(context, listen: false)
+                .ensureInitialized(context);
+          }
+          if (context.mounted) {
+            await Provider.of<ProfileProvider>(context, listen: false)
+                .ensureInitialized(context);
+          }
```

Comment:
Maybe this could be inside a single if statement

Reply:
It can't, because the context may not be on view anymore after the first future completes. The linter actually does not complain if the await is exchange by a callback with then, but I'm guessing the problem would still apply

Reply:
Good catch, didn't think of that :)